google/cloud/aiplatform/jobs.py

```diff
@@ -15,10 +15,214 @@
+                Optional location to retrieve Job subclass from. If not set,
+                location set in aiplatform.init will be used.
+            credentials: Optional[auth_credentials.Credentials] = None,
+                Custom credentials to use. If not set, credentials set in
+                aiplatform.init will be used.
+        """
+        super().__init__(project=project, location=location, credentials=credentials)
+        self.job_subclass_getter_method = getattr(self.api_client, self.getter_method)
+        self._gca_resource = self.job_subclass_getter_method(name=valid_job_name)
+
+    def status(self) -> JobState:
+        """Fetch Job again and return the current JobState.
+
+        Returns:
+            state (JobState):
+                Enum that describes the state of a AI Platform job.
+        """
+
+        # Fetch the Job again for most up-to-date job state
+        self._gca_resource = self.job_subclass_getter_method(
+            name=self._gca_resource.name
+        )
+
+        return self._gca_resource.state
+
+
+class BatchPredictionJob(_Job):
+
+    getter_method = "get_batch_prediction_job"
+
+    def __init__(
+        self,
+        batch_prediction_job_name: str,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """
+        Retrieves a BatchPredictionJob resource and instantiates its representation.
+
+        Args:
+            batch_prediction_job_name (str):
+                Required. A fully-qualified BatchPredictionJob resource name or ID.
+                Example: "projects/.../locations/.../batchPredictionJobs/456" or
+                "456" when project and location are initialized or passed.
+            project: Optional[str] = None,
+                Optional project to retrieve BatchPredictionJob from. If not set,
+                project set in aiplatform.init will be used.
+            location: Optional[str] = None,
+                Optional location to retrieve BatchPredictionJob from. If not set,
+                location set in aiplatform.init will be used.
+            credentials: Optional[auth_credentials.Credentials] = None,
+                Custom credentials to use. If not set, credentials set in
+                aiplatform.init will be used.
+        """
+        valid_batch_prediction_job_name = full_resource_name(
+            resource_name=batch_prediction_job_name,
+            resource_noun="batchPredictionJobs",
+            project=project,
+            location=location,
+        )
+
+        super().__init__(
+            valid_job_name=valid_batch_prediction_job_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+    def iter_outputs(
+        self, bq_query_limit: Optional[int] = 100
+    ) -> Iterable[Union[storage.Blob, bigquery.job.QueryJob]]:
```

Comment:
The output type hint should be `Union[Iterable[storage.Blob], Iterable[bigquery.job.QueryJob]]` based on the docstring description.